### DISCUSSION TITLE
How to use react-query with wordpress rest api?

### DISCUSSION THREAD

**Discussion by rajkinkhabwala** [TanStack/query#2317] [Q&A]
I want to use react-query in react native and I want to fetch posts from WordPress. Actually, the posts are getting fetched easily using the rest endpoint of any WordPress blog. But the main issue is that while fetching those posts I also want to fetch the WordPress featured image and the author name and for that I have to again do a request to the server for fetching the data. But I dont know how I can proceed for this requirement for mine.

**Comment by TkDodo** [comment]
looks like you want dependent queries? have a look here:

https://react-query.tanstack.com/guides/dependent-queries

**Reply by rajkinkhabwala** [reply]
hey, I have looked into that, but I was not able to get the result I want. Is it possible for you that you can provide me with a  demo code?

**Reply by TkDodo** [reply]
I think it's the other way around. If you come up with a demo code (codesandbox preferably), I can take a look :)

**Reply by rajkinkhabwala** [reply]
https://codesandbox.io/s/sleepy-forest-mz27s
hey, check this out I have added a small example code from which I want to fetch data. please do help me out with the code.

**Reply by TkDodo** [reply]
There are a couple of things not quite right here:

- falsy query keys
- queryClient created _inside_ App.js
- variables not passed correctly to queryFn
- anonymous inline components (`AllData`) that will be mounted on every render
- not handling errors / accessing data without checking for it first

when all that is fixed, what remains is a CORS error (AllowOriginMismatch) when trying to fetch from `https://www.dimensionplus.in/wp-json/wp/v2/posts/` which is something that you will have to look into.

see: https://codesandbox.io/s/hungry-bohr-l5yne?file=/src/App.js
